### PR TITLE
Enable status bar for notebooks

### PR DESCRIPTION
**Pull request by lucyzhang929**
This PR fixes https://github.com/microsoft/azuredatastudio/issues/14369

### CODE REVIEW

**Review by alanrenmsft — APPROVED** [review]

File: src/sql/workbench/services/notebook/browser/notebookService.ts [hunk 1]
```diff
@@ -129,6 +129,16 @@ export interface INotebookService {
 	 * @param isTrusted True if notebook is to be set to trusted, false otherwise.
 	 */
 	setTrusted(notebookUri: URI, isTrusted: boolean): Promise<boolean>;
+
+	/**
+	 * Event that gets fired when a new Sql query is started.
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/notebook/browser/notebookService.ts:134]: It seems weird to have SQL specific events in the service API...is there any reason you couldn't just have this fire for all code cell executions and update the names to be more generic?

File: src/sql/workbench/contrib/query/browser/statusBarItems.ts [hunk 2]
```diff
@@ -289,7 +292,7 @@ export class QueryResultSelectionSummaryStatusBarContribution extends Disposable
 
 	private onCellSelectionChanged(data: string[]): void {
 		const numericValues = data?.filter(value => !Number.isNaN(Number(value))).map(value => Number(value));
-		if (numericValues?.length < 2 || !(this.editorService.activeEditor instanceof QueryEditorInput)) {
+		if (numericValues?.length < 2) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/query/browser/statusBarItems.ts:295]: So does this already handle displaying the status bar item for Notebook editors with the input check being removed? And then you just also added the call to hide it when the query is started?

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]

File: src/sql/workbench/services/notebook/browser/notebookService.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/services/notebook/browser/notebookService.ts:134]: @chlafreniere thoughts on this?

**Comment by Charles-Gagnon** [comment]
Also how does this work when you have multiple cells with output? Will it update based on the output selected - and then if you unselect that grid what happens to the status bar item?

This stuff can probably wait for the feature to get in and then have some people just bash on it but it'd be good to play around with it now and see if there's any weird behavior so we can decide whether we want to fix it or just mart it as expected behavior/known issue.

File: src/sql/workbench/contrib/query/browser/statusBarItems.ts (hunk 2 above)
Comment (lucyzhang929) [reply src/sql/workbench/contrib/query/browser/statusBarItems.ts:295]: Yep I believe the input check was added temporarily to disable the select summary status bar for notebooks. Is that correct @alanrenmsft?

File: src/sql/workbench/services/notebook/browser/notebookService.ts (hunk 1 above)
Comment (lucyzhang929) [reply src/sql/workbench/services/notebook/browser/notebookService.ts:134]: @Charles-Gagnon you're right, having something more generic like onCodeCellExecutionStart event would make more sense.

**Comment by lucyzhang929** [comment]
@Charles-Gagnon The status bar will update based on which table is selected and unselecting the grid doesn't remove the status bar item. This is the same behavior as query editor since notebooks and query editor share the same GridTableBase. Agree that some testing would be helpful. 

@chlafreniere To answer your earlier question, the status bar summary will still work when we open a saved notebook with tables since the data is saved in our DataResourceDataProvider class.

File: src/sql/workbench/contrib/notebook/test/stubs.ts [hunk 3]
```diff
@@ -289,6 +289,12 @@ export class NotebookServiceStub implements INotebookService {
 	navigateTo(notebookUri: URI, sectionId: string): void {
 		throw new Error('Method not implemented.');
 	}
+	get onCellExecutionStart(): vsEvent.Event<void> {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/test/stubs.ts:292]: `onCodeCellExecutionStart` perhaps?

**Review by Charles-Gagnon — APPROVED** [review]

**Review by chlafreniere — APPROVED** [review]

**Pull request merged**